src/Microsoft.CodeQuality.Analyzers/Core/QualityGuidelines/AvoidDuplicateElementInitialization.cs

```diff
@@ -0,0 +1,133 @@
+                                                                             s_localizableMessage,
+                                                                             DiagnosticCategory.Usage,
+                                                                             DiagnosticHelpers.DefaultDiagnosticSeverity,
+                                                                             isEnabledByDefault: true,
+                                                                             description: s_localizableDescription,
+                                                                             helpLinkUri: null,     // TODO: add MSDN url
+                                                                             customTags: WellKnownDiagnosticTags.Telemetry);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        private static readonly ConcurrentDictionary<string, SyntaxGenerator> s_generators = new ConcurrentDictionary<string, SyntaxGenerator>();
+
+        public override void Initialize(AnalysisContext analysisContext)
+        {
+            analysisContext.EnableConcurrentExecution();
+            analysisContext.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            analysisContext.RegisterOperationAction(AnalyzeOperation, OperationKind.ObjectCreation);
+        }
+
+        private static void AnalyzeOperation(OperationAnalysisContext context)
+        {
+            var objectInitializer = (IObjectCreationOperation)context.Operation;
+            if (objectInitializer.Initializer == null)
+            {
+                return;
+            }
+
+            var initializedElementIndexes = new HashSet<object[]>(ConstantArgumentEqualityComparer.Instance);
+
+            foreach (var initializer in objectInitializer.Initializer.Initializers)
+            {
+                if (initializer is ISimpleAssignmentOperation assignment &&
+                    assignment.Target is IPropertyReferenceOperation propertyReference &&
+                    propertyReference.Arguments.Length != 0)
+                {
+                    var values = GetConstantArgumentValues(propertyReference.Arguments);
+                    if (values != null && !initializedElementIndexes.Add(values))
+                    {
+                        var generator = s_generators.GetOrAdd(
+                            context.Compilation.Language,
+                            language => SyntaxGenerator.GetGenerator(new AdhocWorkspace(), language));
+                        var indexesText = string.Join(", ", values.Select(value => generator.LiteralExpression(value)));
+                        context.ReportDiagnostic(
+                            Diagnostic.Create(Rule, propertyReference.Syntax.GetLocation(), indexesText));
+                    }
+                }
+
+                if (context.CancellationToken.IsCancellationRequested)
```

Comment:
Not required. It is recommended to check for cancellation in a loop only if you are doing very expensive operations.

Reply:
Done